rclpy/rclpy/node.py

```diff
@@ -349,6 +349,9 @@ def declare_parameter(
         This method, if successful, will result in any callback registered with
         :func:`add_on_set_parameters_callback` to be called.
 
+        The name and type in the given descriptor is ignored, and should be specified using
+        the name argument to this function and the default value's type instead.
```

Comment:
Is this true? It's valid to provide no default value and a descriptor, in which case I presume it would infer the type from the descriptor.

Also, this change seems unrelated to the async parameter client feature.

Reply:
I think this is something that snuck in from the revert.

Reply:
Line was introduced here: https://github.com/ros2/rclpy/commit/b90edf080f3326ef06323f3a41ccccf40b493cee

Reply:
Nevermind, it looks like this change is on `master`, it's probably just reappearing here because it was recommitted on your branch with a different hash.